lib/ex_unit/test/ex_unit/case_test.exs

```diff
@@ -262,3 +262,55 @@ defmodule ExUnit.BadOptsCase do
     end
   end
 end
+
+defmodule ExUnit.CaseTest.MostRecentRegisteredTestHelper do
+  defmacro escaped_most_recent_registered_test do
+    Macro.escape(ExUnit.Case.most_recent_registered_test(__CALLER__))
+  end
+end
+
+defmodule ExUnit.CaseTest.MostRecentRegisteredTestTest do
+  use ExUnit.Case, async: true
+  import ExUnit.CaseTest.MostRecentRegisteredTestHelper
+
+  most_recent = ExUnit.Case.most_recent_registered_test(__MODULE__)
+
+  test "should return nil if called before any test has been registered" do
```

Comment:
nitpick: i don't think we use "should" prefix in our tests. Rather `test "returns nil...`.